logic/resolvable/Concludable.java

```diff
@@ -244,21 +242,21 @@ public Relation(final RelationConstraint constraint, @Nullable IsaConstraint isa
 
         @Override
         public ResourceIterator<Unifier> unify(Rule.Conclusion.Relation relationConclusion, ConceptManager conceptMgr) {
-            if (this.constraint().players().size() > relationConclusion.constraint().players().size())
+            if (this.constraint().players().size() > relationConclusion.relation().players().size())
                 return Iterators.empty();
             Unifier.Builder unifierBuilder = Unifier.builder();
 
             if (!constraint().owner().reference().isAnonymous()) {
                 assert constraint().owner().reference().isName();
-                if (unificationSatisfiable(constraint().owner(), relationConclusion.constraint().owner())) {
-                    unifierBuilder.add(constraint().owner().id(), relationConclusion.constraint().owner().id());
+                if (unificationSatisfiable(constraint().owner(), relationConclusion.relation().owner())) {
+                    unifierBuilder.add(constraint().owner().id(), relationConclusion.relation().owner().id());
                 } else return Iterators.empty();
             }
 
             if (constraint().owner().isa().isPresent()) {
-                assert relationConclusion.constraint().owner().isa().isPresent(); // due to known shapes of rule conclusions
+                assert relationConclusion.relation().owner().isa().isPresent(); // due to known shapes of rule conclusions
```

Comment:
i think we can get rid of this now, as the assertion on the "shape" of each conclusion will live in the conclusion constructors

Reply:
Ah yes, thought this was gone.